src/Physics/Evaluate_friction_law.f90

```diff
@@ -1607,4 +1626,106 @@ SUBROUTINE rate_and_state_nuc103(TractionGP_XY,TractionGP_XZ,        & ! OUT: tr
 
  END SUBROUTINE rate_and_state_nuc103
 
+   SUBROUTINE update_RSF (IO, nBndGP, RS_f0, RS_b, RS_a, RS_sr0, RS_fw, RS_srW, RS_sl0, &
+                         SV0, time_inc, SR_tmp, LocSR, LocSV)
+    !-------------------------------------------------------------------------!
+    IMPLICIT NONE
+    !-------------------------------------------------------------------------!
+    ! Argument list declaration
+    TYPE(tInputOutput)       :: IO
+    INTEGER                  :: nBndGP
+    REAL                     :: RS_f0, RS_b, RS_a(nBndGP), RS_sr0, RS_fw, RS_srW(nBndGP), RS_sl0(nBndGP) !constant input parameters
+    REAL                     :: SV0(nBndGP), time_inc, SR_tmp(nBndGP), LocSR(nBndGP)                  !changing during iterations
+    REAL                     :: flv(nBndGP), fss(nBndGP), SVss(nBndGP), LocSV(nBndGP)                 !calculated in this routine
+    !-------------------------------------------------------------------------!
+    INTENT(IN)    :: IO, RS_f0, RS_b, RS_a, RS_sr0, RS_fw, RS_srW, &
+                     RS_sl0, SV0, time_inc, SR_tmp, LocSR
+    INTENT(INOUT) :: LocSV
+    !-------------------------------------------------------------------------!
+
+    ! low-velocity steady state friction coefficient
+    flv = RS_f0 - (RS_b-RS_a)* LOG(SR_tmp/RS_sr0)
+    ! steady state friction coefficient
+    fss = RS_fw + (flv - RS_fw)/(1.0D0+(SR_tmp/RS_srW)**8d0)**(1.0D0/8.0D0)
+    ! steady-state state variable
+    ! For compiling reasons we write SINH(X)=(EXP(X)-EXP(-X))/2
+    SVss = RS_a * LOG(2.0D0*RS_sr0/SR_tmp * (EXP(fss/RS_a)-EXP(-fss/RS_a))/2.0D0)
+
+    ! exact integration of dSV/dt DGL, assuming constant V over integration step
+    LocSV = Svss*(1.0D0-EXP(-SR_tmp*time_inc/RS_sl0))+EXP(-SR_tmp*time_inc/RS_sl0)*SV0
+
+
+    IF (ANY(IsNaN(LocSV)) .EQV. .TRUE.) THEN
+       STOP
```

Comment:
add something like logError(*) 'NaN detected', time?

Reply:
this is already in the master branch with Nan detected. I haven't pulled it yet.